timemachine/fe/single_topology.py

```diff
@@ -1615,6 +1471,315 @@ def _setup_intermediate_chiral_bond_term(
+        return any([x in dummy_atoms for x in idxs])
+
+    def any_idxs_belong_to_dummy_b(self, idxs):
+        dummy_atoms = self.get_dummy_atoms_b()
+        return any([x in dummy_atoms for x in idxs])
+
+    def _find_core_chiral_atoms_impl(self, src_idxs, dst_idxs, dst_params, filter_fn):
+        src_chiral_idxs = set(tuple(x) for x in src_idxs)
+        dst_chiral_idxs = set(tuple(x) for x in dst_idxs)
+        chiral_atoms_turning_on = dst_chiral_idxs.difference(src_chiral_idxs)
+
+        src_chiral_atom_idxs = np.zeros((0, 4))  # empty by construction
+        src_chiral_atom_params = np.zeros([])  # empty by construction
+        src_chiral_atom = ChiralAtomRestraint(src_chiral_atom_idxs).bind(src_chiral_atom_params)
+
+        dst_chiral_atom_idxs = []
+        dst_chiral_atom_params = []
+        for idxs, params in zip(dst_idxs, dst_params):
+            idxs = tuple(idxs)
+            if idxs in chiral_atoms_turning_on and filter_fn(idxs):
+                dst_chiral_atom_idxs.append(idxs)
+                dst_chiral_atom_params.append(params)
+
+        dst_chiral_atom = ChiralAtomRestraint(np.array(dst_chiral_atom_idxs)).bind(dst_chiral_atom_params)
+        return src_chiral_atom, dst_chiral_atom
+
+    def find_core_chiral_atoms_turning_on(self):
+        return self._find_core_chiral_atoms_impl(
+            self.src_system.chiral_atom.potential.idxs,
+            self.dst_system.chiral_atom.potential.idxs,
+            self.dst_system.chiral_atom.params,
+            self.all_idxs_belong_to_core,
+        )
+
+    def find_core_chiral_atoms_turning_off(self):
+        x, y = self._find_core_chiral_atoms_impl(
+            self.dst_system.chiral_atom.potential.idxs,
+            self.src_system.chiral_atom.potential.idxs,
+            self.src_system.chiral_atom.params,
+            self.all_idxs_belong_to_core,
+        )
+        return y, x
+
+    def find_core_chiral_atoms_unaffected(self):
+        src_chiral_idxs = set(tuple(x) for x in self.src_system.chiral_atom.potential.idxs)
+        dst_chiral_idxs = set(tuple(x) for x in self.dst_system.chiral_atom.potential.idxs)
+        maintained_idxs = dst_chiral_idxs.intersection(src_chiral_idxs)
+
+        chiral_atom_idxs = []
+        chiral_atom_params = []
+        for idxs, params in zip(self.src_system.chiral_atom.potential.idxs, self.src_system.chiral_atom.params):
+            idxs = tuple(idxs)
+            if idxs in maintained_idxs and self.all_idxs_belong_to_core(idxs):
+                chiral_atom_idxs.append(idxs)
+                chiral_atom_params.append(params)
+
+        chiral_atom_idxs = np.array(chiral_atom_idxs).reshape(-1, 4)
+        chiral_atom_params = np.array(chiral_atom_params).reshape(-1)
+
+        src_chiral_atom = ChiralAtomRestraint(chiral_atom_idxs).bind(chiral_atom_params)
+        dst_chiral_atom = ChiralAtomRestraint(chiral_atom_idxs).bind(chiral_atom_params)
+
+        return src_chiral_atom, dst_chiral_atom
```

Comment:
unclear intent: `find_core_chiral_atoms_unaffected` returns two identical objects (and the three functions `find_core_chiral_atoms_{unaffected/turning_on/turning_off}` appear not to be referenced currently)

Reply:
sorry stale code from a bad merge earlier - will remove the dead code